framework/cancelercontext/canceler_context.go

```diff
@@ -0,0 +1,47 @@
+// Package cancelercontext stores and accesses the HTTP Authorization precondition
+// context.Context.
+package cancelercontext
```

Comment:
Here we add a canceler context package that enables passing a canceler channel around within a context.

Reply:
Looks a bit like reinventing the wheel. Is there a reason why WithCancel can't be used? https://golang.org/pkg/context/#WithCancel

Reply:
Please read my Slack monologue linked in the description.